### PR TITLE
Authentication

### PR DESCRIPTION
**Pull request by lars-vc**
Added authentication and some tests.
## Auth
Added example on how to use auth in readme
## Tests
The most reliable way to currently run the tests is by running the script: `./run_tests.sh` or using npm: `npm run test2`

### CODE REVIEW

**Comment by lars-vc** [comment]
The componentsjs issues still exists

File: src/ldp/LDPCommunication.ts [hunk 1]
```diff
@@ -11,11 +11,12 @@ export class LDPCommunication implements Communication {
     public constructor() {
     }
 
+
     public async get(resourceIdentifier: string, headers?: Headers): Promise<Response> {
         headers = headers ? headers : new Headers({'Accept': 'text/turtle'})
         return await fetch(resourceIdentifier, {
             method: 'GET',
-            headers
+            headers: headers
```
Comment (woutslabbinck) [inline src/ldp/LDPCommunication.ts:19]: I don't see the benefit of this change

File: test/global-setup.ts [hunk 2]
```diff
@@ -1,19 +1,20 @@
...
 
 async function start(): Promise<void> {
-  // start server and wait till it is running + login and wait till that has succeeded
-  runSolid();
-  await isRunning();
-
+    // start server and wait till it is running + login and wait till that has succeeded
+    await runSolid();
+    // await runAuthSolid();
+    // await createTestAcc();
```
Comment (woutslabbinck) [inline test/global-setup.ts:7]: Can be removed, right?

File: test/util/solidHelper.ts [hunk 3]
```diff
@@ -1,17 +1,52 @@
...
+        applicationName: "LDES-orchestrator",
+        registrationType: RegistrationType.Dynamic,
+        solidIdentityProvider: authBaseUrl
+    };
+    console.log("Login with email: test@mail.com password: test");
+    await login(validatedOptions);
+    await isLoggedin(); // code that checks whether you are already logged in
+}
+
+export async function createTestAcc() {
```
Comment (woutslabbinck) [inline test/util/solidHelper.ts:33]: Isn't this done by the test script now?

File: src/versionawarelil/VersionAwareLDESinLDP.ts [hunk 4]
```diff
@@ -15,6 +15,7 @@ import {extractLdesMetadata, LDESMetadata} from "../util/LdesUtil";
 import {addDeletedTriple, addVersionSpecificTriples, isDeleted, removeVersionSpecificTriples} from "./Util";
 import namedNode = DataFactory.namedNode;
 
+
```
Comment (woutslabbinck) [inline src/versionawarelil/VersionAwareLDESinLDP.ts:18]: Strange addition: I'm personally more a single newline guy tho :wink:

**Review by woutslabbinck — CHANGES_REQUESTED** [review]
Nice work!
I have some small suggestions, but it looks very good!

File: example.js [hunk 5]
```diff
@@ -0,0 +1,32 @@
+async function f() {
```
[the remark this answers is not in the document]
Comment (lars-vc) [reply example.js:1]: Ah I thought a full example file wouldn't hurt, but I understand if you don't want it

[the remark this answers is not in the document]
Comment (woutslabbinck) [reply example.js:1]: I mean, we have the readme for that

File: src/ldp/LDPCommunication.ts (hunk 1 above)
Comment (lars-vc) [reply src/ldp/LDPCommunication.ts:19]: Consistency with all the other functions and in my personal opinion Im not a huge fan of that syntax, it's a bit cryptic imo.

File: test/util/solidHelper.ts (hunk 3 above)
Comment (lars-vc) [reply test/util/solidHelper.ts:33]: This `initAuth` function is still needed, but the createTestAcc is replaced by the script jes,

Comment (woutslabbinck) [reply test/util/solidHelper.ts]: Oh, I did not know that. Then only one of them can be remove.

**Review by woutslabbinck — APPROVED** [review]

**Pull request merged**